source-git-monorepos/README.md

```diff
@@ -50,21 +89,367 @@ packages:
+  dist_git_path: rpms/copr-backend
+  dist_git_branch: rawhide
+- downstream_package_name: copr-frontend
+  package_path: frontend
+  dist_git_path: rpms/copr-frontend
+  dist_git_branch: rawhide
+- downstream_package_name: copr-cli
+  package_path: cli
+  dist_git_path: rpms/copr-cli
+  dist_git_branch: rawhide
+```
+
+Which could be used as:
+
+```shell
+$ packit source-git init --packages packages.yaml v3.0.0 src/copr
+```
+
+Or the same could be read from STDIN, with `--packages=-`:
+
+```shell
+$ packit source-git init --packages=- v3.0.0 src/copr
+```
+
+Given that setting up a source-git repository is expected to be a one-time
+activity, and that there are not so many projects which are monorepos,
+implementing the above should probably have **a low priority for now**.
+
+### `packit source-git update-dist-git`
+
+**At a minimum**, the command should be updated to support monorepos, and
+transform only the content specific to the package from source-git. The
+package name (`downstream_package_name`) to be considered from
+`.distro/source-git.yaml` is the one which matches `basename DIST_GIT`.
 
-For **backwards compatibility**: if the top-level `packages` key is missing,
+**Going further**, when used with monorepos, this command could be able to
+update multiple dist-git repositories. This means, that the `DIST_GIT`
+argument of the command would become `DIST_GIT [DIST_GIT...]` (one or more
+paths to dist-git repositories).
+
+In the form above, the command would update only the dist-git repositories
+which are specified on the command line. If any of the `DIST_GIT` arguments
+specified is not a `downstream_package_name` in `.distro/source-git.yaml`, the
+command would fail. This would be an early check, before any other change is
+done.
+
+An alternative form of the command could allow updating multiple dist-git
+repositories found in `--dist-git-store`. The command would figure out
+dist-git subdirectory names in `--dist-git-store`, based on
+`downstream_package_name`s found in `.distro/source-git.yaml`. If any of them
+is missing, the command would fail early.
+
+```
+SYNOPSIS
+       packit source-git update-dist-git [OPTIONS] --dist-git-store PATH SOURCE_GIT
+```
+
+Example:
+
+```shell
+$ packit source-git update-dist-git --dist-git-store rpms/ src/copr
+```
+
+### `packit source-git update-source-git`
+
+Let's not change the interface of this command and don't support the usecase
+to update a single source-git monorepo from multiple dist-git repositories in
+order to avoid complications. Expect that multiple commands are going to be
+executed if updates need to be taken from multiple dist-git repositories.
+
+On the other hand, the command needs to be updated to support a source-git
+repo as a target when the source-git repo is a monorepo. In this case only the
+paths of the sub-package need to be considered during the update. `DIST_GIT`
+should match one of the `downstream_package_name`s in `.distro/source-git.yaml`.
```

Comment:
This command could IMO be called also from within the source-git subdirectory.

Reply:
This command could be called from any path as it'll receive 3 arguments: the path to the dist-git repository, the path to the source-git repo, and a revision range in dist-git.